Algorithm/QCAlgorithm.Universe.cs

```diff
@@ -419,6 +419,30 @@ public void AddUniverse(SecurityType securityType, string name, Resolution resol
             AddUniverse(new UserDefinedUniverse(config, universeSettings, SecurityInitializer, resolution.ToTimeSpan(), selector));
         }
 
+        /// <summary>
+        /// Adds the benchmark security to the benchmark universe
+        /// </summary>
+        private void AddToBenchmarkUniverse(Security security)
+        {
+            var universeSymbol = BenchmarkUniverse.CreateSymbol(security.Type, security.Symbol.ID.Market);
+
+            Universe universe;
+            if (!UniverseManager.TryGetValue(universeSymbol, out universe))
+            {
+                var objectType = SubscriptionManager.LookupSubscriptionConfigDataTypes(security.Type, security.Resolution, false).First().Item1;
+                var marketHoursDbEntry = MarketHoursDatabase.GetEntry(security.Symbol.ID.Market, security.Symbol.Value, security.Type);
+                var dataTimeZone = marketHoursDbEntry.DataTimeZone;
+                var exchangeTimeZone = marketHoursDbEntry.ExchangeHours.TimeZone;
+                var config = new SubscriptionDataConfig(objectType, universeSymbol, security.Resolution, dataTimeZone, exchangeTimeZone, false, false, true, isFilteredSubscription: false);
+                universe = new BenchmarkUniverse(config);
+
+                UniverseManager.Add(universeSymbol, universe);
+            }
+
+            var benchmarkUniverse = universe as BenchmarkUniverse;
+            benchmarkUniverse?.AddBenchmarkSecurity(UtcTime, security);
```

Comment:
we should remove the `?` here -- if it's not a `BenchmarkUniverse` then I think it's correct to throw, perhaps something like this:
``` csharp
if (benchmarkUniverse == null)
{
    throw new Exception($"Expected BenchmarkUniverse with symbol {universeSymbol} but received {univerese.GetType().FullName}");
}
benchmarkUniveres.AddBenchmarkSecurity(UtcTime, security);
```